plutus-ledger-api/src/Plutus/V1/Ledger/Tx.hs

```diff
@@ -321,28 +319,11 @@ pubKeyTxIn r = TxIn r ConsumePublicKeyAddress
 scriptTxIn :: TxOutRef -> Validator -> Redeemer -> Datum -> TxIn
 scriptTxIn ref v r d = TxIn ref $ ConsumeScriptAddress v r d
 
--- | The type of a transaction output.
-data TxOutType =
-    PayToScript !DatumHash -- ^ A pay-to-script output with the given datum hash.
-    | PayToPubKey -- ^ A pay-to-pubkey output.
-    deriving stock (Show, Eq, Ord, Generic)
-    deriving anyclass (Serialise, ToJSON, FromJSON, ToJSONKey, NFData)
-
-instance PlutusTx.Eq TxOutType where
-    PayToScript l == PayToScript r = l PlutusTx.== r
-    PayToPubKey == PayToPubKey     = True
-    _ == _                         = False
-
-instance Pretty TxOutType where
-    pretty = \case
-        PayToScript ds -> "PayToScript:" <+> pretty ds
-        PayToPubKey    -> "PayToPubKey"
-
--- | A transaction output, consisting of a target address, a value, and an output type.
+-- | A transaction output, consisting of a target address, a value, and optionally a datum hash.
 data TxOut = TxOut {
-    txOutAddress :: !Address,
-    txOutValue   :: !Value,
-    txOutType    :: !TxOutType
+    txOutAddress   :: !Address,
```

Comment:
get rid of the bangs. Although maybe they help us with the off-chain version, in which case disregard all these comments...

Reply:
You get a warning about a strict field not being initialised if you forget it... not sure if that's worth it. I doubt it has any performance implications for us (although I never really looked into it, so I could be wrong)